### ISSUE TITLE
Should change Label type from str to int

### ISSUE THREAD

**Issue by eddyxu** [eto-ai/rikai#46]
So that Label can be also applied to the inferred results directly.

**Comment by eddyxu** [comment]
wdyt @changhiskhan

**Comment by eddyxu** [comment]
We can represent today's `Label` to a `Category` or `Class`

**Comment by changhiskhan** [comment]
So Label == StringIndexer(Category) ? I think it can make sense as long as the use case is clear and we can provide clear documentation / conveniences to convert between the two.

**Comment by eddyxu** [comment]
Yes. For example, in batch inference code, the returned values are always numeric values. it is nice to build strong typed output from it.

**Comment by eddyxu** [comment]
@sadhen would like to hear your opinions on this.  In general, we'd like that `rikai` dataset has strong types. That is, while two fields might be the same `Integer` or `String` internally, however, rikai acks that they have a particular meaning in ML setup. 

For example, a `label / class` integer here can help us to build tooling to run aggregation on top of it, because we know there are limited value range of this value, while in the meantime, the value is important for us to analyze the distribution, balance of dataset and etc.
